aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -800,7 +800,7 @@ Tensor add_sparse_csr(
     const Scalar& alpha) {
   auto commonDtype = at::result_type(self, other);
   alpha_check(commonDtype, alpha);
-  Tensor result = at::empty({0, 0}, self.options().dtype(commonDtype));
+  Tensor result = at::empty_like(self, self.options().dtype(commonDtype));
```

Comment:
That was really annoying. `options()` do not store any index dtype...

Reply:
index dtype is not an option for sparse tensor constructors. I would expect that `empty_like` would use `self` indices to set its indices dtype.